src/AppBundle/Controller/AutomatedEditsController.php

```diff
@@ -0,0 +1,249 @@
+            return $this->redirectToRoute(
+                "autoeditsResult",
+                [
+                    'project'=>$projectQuery
+                ]
+            );
+        }
+
+        // set default wiki so we can populate the namespace selector
+        if (!$projectQuery) {
+            $projectQuery = $this->container->getParameter('default_project');
+        }
+
+        /** @var ApiHelper */
+        $api = $this->get("app.api_helper");
+
+        return $this->render('autoEdits/index.html.twig', [
+            'xtPageTitle' => 'tool_autoedits',
+            'xtSubtitle' => 'tool_autoedits_desc',
+            'xtPage' => 'autoedits',
+            'project' => $projectQuery,
+            'namespaces' => $api->namespaces($projectQuery),
+        ]);
+    }
+
+    /**
+     * @Route("/autoedits/{project}/{username}/{begin}/{end}", name="autoeditsResult")
+     */
+    public function resultAction($project, $username, $begin = null, $end = null)
+    {
+        // Pull the labs helper and check if enabled
+        $lh = $this->get("app.labs_helper");
+        $lh->checkEnabled("autoedits");
+
+        // Pull information about the project from the Labs Helper
+        $dbValues = $lh->databasePrepare($project, "AutomatedEdits");
+
+        $dbName = $dbValues["dbName"];
+        $wikiName = $dbValues["wikiName"];
+        $url = $dbValues["url"];
+
+        // Grab our database connection
+        $dbh = $this->get('doctrine')->getManager("replicas")->getConnection();
+
+        // Variable parsing.
+        // Username needs to be uppercase first (yay Mediawiki),
+        // and we also need to handle undefined dates.
+        $username = ucfirst($username);
+
+        if ($begin == null) {
+            $begin = date("Y-m-d", strtotime("-1 month"));
+        }
+
+        if ($end == null) {
+            $end = date("Y-m-d");
+        }
+
+        // Validating the dates.  If the dates are invalid, we'll redirect
+        // to the project and username view.
+        if (strtotime($begin) === false || strtotime($end) === false) {
+            // Make sure to add the flash notice first.
+            $this->addFlash("notice", ["invalid_date"]);
+
+            // Then redirect us!
+            return $this->redirectToRoute(
+                "autoeditsResult",
+                [
+                    "project" => $project,
+                    "username" => $username,
+                ]
+            );
+        }
+
+        // Now, load the semi-automated edit types.
+        $AEBTypes = [];
+        $AEBTypes = $this->getParameter("automated_tools");
+
+        // Create a collection of queries that we're going to run.
+        $queries = [];
+
+        $rev = $lh->getTable("revision", $dbName);
+        $arc = $lh->getTable("archive", $dbName);
+  
+        $cond_begin = ( $begin ) ? " AND rev_timestamp > '$begin' " : null;
+        $cond_end = ( $end ) ? " AND rev_timestamp < '$end' ": null;
```

Comment:
These should use named parameters, rather than string construction. Especially as this is end-user input. (I know they're validated to satisfy strtotime above, but I think they should still be parameters.)